lib/logitech_receiver/receiver.py

```diff
@@ -99,8 +105,15 @@ def __init__(self, receiver, number, link_notification=None):
 				# unifying protocol not supported, must be a Nano receiver
 				device_info = self.receiver.read_register(_R.receiver_info, 0x04)
 				if device_info is None:
-					_log.error("failed to read Nano wpid for device %d of %s", number, receiver)
-					raise _base.NoSuchDevice(number=number, receiver=receiver, error="read Nano wpid")
+					#ex100 receiver, fill fake device_info with known wpid's
+					if ((receiver.product_id == 'c517') and ((number==1) or (number==3))):
+						if (number == 1):#mouse
+							device_info = '   '+'\x3F\x21'
+						if (number == 3): #keyboard
+							device_info = '   '+'\x65\x11'
```

Comment:
Have you tested this with Python 3? It should probably be `b'...'`. Even better, just set wpid directly? E.g.:

```python
if pair_info:
    ...
elif receiver.product_id == 'c517' and number in [1, 3]:
    ... # ex100 logic
else:
   ... # nano receiver logic
```

Reply:
First, sorry for long delay, looks like I missed mail about your comment.
1: Not expert in Python, your mean use something like `device_info = b('\x20\x20\x20\x3F\x21')` ?
2: Not sure it is good idea/ Currently, logic is as I understand 

read receiver info register
if pair_info:
   fill data
else: 
  read nano receiver register
  if device_info is None:
     both registers failed, so it nether a unifying nor nano receiver, try fall back for ex100

And you propose first check for unifying, next check if it is ex100 and last try nano register?

Reply:
> And you propose first check for unifying, next check if it is ex100 and last try nano register?

Yes that was the suggestion, but it would then assume that this `receiver.product_id` is only used for the EX100. If that is not the case, perhaps the proposed approach here would be fine.

As for the Python 3-compatible version for your if statements:
```py
if number == 1:  # mouse
    device_info = b'\x20\x20\x20\x3f\x21'
if number == 3:  # keyboard
    device_info = b'\x20\x20\x20\x65\x11'
```